openedx/core/djangoapps/content_libraries/tasks.py

```diff
@@ -477,3 +484,66 @@ def _copy_overrides(
                 dest_block=store.get_item(dest_child_key),
             )
     store.update_item(dest_block, user_id)
+
+
+class LibraryBackupTask(UserTask):  # pylint: disable=abstract-method
+    """
+    Base class for tasks related with Library backup functionality.
+    """
+
+    @classmethod
+    def generate_name(cls, arguments_dict) -> str:
+        """
+        Create a name for this particular backup task instance.
+
+        Should be both:
+        a. semi human-friendly
+        b. something we can query in order to determine whether the library has a task in progress
+
+        Arguments:
+            arguments_dict (dict): The arguments given to the task function
+
+        Returns:
+            str: The generated name
+        """
+        key = arguments_dict['library_key_str']
+        return f'Export of {key}'
```

Comment:
```suggestion
        return f'Backup of {key}'
```

"Export" is going to be a more fine-grained thing that we'll work out later, and will likely include partial exports (e.g. "export this collection" or "export  this subsection and all its descendants").

Reply:
done, thanks!